src/core/layout/qgslayoutitempolygon.cpp

```diff
@@ -129,6 +129,23 @@ QgsGeometry QgsLayoutItemPolygon::clipPath() const
   return QgsGeometry::fromQPolygonF( path );
 }
 
+
+bool QgsLayoutItemPolygon::isValid() const
+{
+  QList<QPointF> uniquePoints;
```

Comment:
Maybe a little comment to ease readability here to tell that the polygon is valid is there is at least 3 different points

Reply:
That would indeed be bettter.